suite/tests/api/ir_x86.c

```diff
@@ -787,6 +787,98 @@ test_instr_encode_and_decode(void *dc, instr_t *instr, uint len_expect,
     instr_destroy(dc, decin);
 }
 
+bool
+instr_same_synthetic(instr_t *instr_orig, instr_t *instr_copy)
```

Comment:
Better to have instr_same() handle synthetic ISA instrs.  Once in instr_t form they should look like other instrs: maybe just add isa mode and category checks to instr_same()?  Anything else needed?

Reply:
Mhm, there isn't much in common between the `instr_same()` and `instr_same_synthetic()` that's why I separeted them.
We can handle both in just `instr_same()` but it would look awfully close to:
```
instr_same()
{
   if (DR_ISA_SYNTHETIC) {
       do what instr_same_synthetic does ...
   } else {
       do what instr_same() does ...
   }
}
```
The purpose of `instr_same_synthetic()` is to check that an instr_t from a real architecture (e.g., x86, ARM) has the same "invariants" (for lack of a better word) of an instr_t that comes from our synthetic encoding.
For example, we cannot check the number of source operands of both instr_t using `instr_num_srcs()` as `instr_same()` does, because if the real instr_t had a memory reference destination operand with a register, the synthetic instr_t will have that as an additional source operand, so they wouldn't match, even though they should.
Also, we cannot use `opnd_same()` like `instr_same()` does, because a synthetic instr_t only has registers.
Other instr_t fields are simply lost in the synthetic ISA, like the opcode, so we cannot use `instr_get_opcode()` like `instr_same()` does.

Reply:
Ah, I thought this was comparing 2 synthetic instrs.  Yes, the name was part of the issue (plus tending to spend less time reviewing code in tests).  Can we rename it to instr_synthetic_matches_real() or some such?

Reply:
Done.